runtime/router/trie.go

```diff
@@ -140,8 +151,7 @@ func (t *tnode) set(path string, value http.Handler, lastKeyCharSlash, lastPathC
 	// is immediately after slash, e.g. "/:foo", "/x/:y". "/a:b" is not a colon wildcard segment.
 	var keyMatchIdx, pathMatchIdx int
 	for keyMatchIdx < keyLength && pathMatchIdx < pathLength {
-		if (t.key[keyMatchIdx] == ':' && lastKeyCharSlash) ||
-			(path[pathMatchIdx] == ':' && lastPathCharSlash) {
+		if t.isWildCardPattern(path, keyMatchIdx, pathMatchIdx, lastKeyCharSlash, lastPathCharSlash, false, isWhitelisted) {
```

Comment:
Nope, we need to change the logic other wise it will not work for below sequence of paths for whitelisted paths
/a/b
/a/:var
Without the logic that I added the second path handler will replace the value of the /a/b node.

Reply:
sure, but can we make a specific function for getter and setter so we know which path is needed and which is redundant. makes it easier to maintain.

Reply:
The logic is same for both and it is controlled by colonAsPattern flag. This flag is not something that I introduced.

Reply:
Created separate functions for getter and setter